src/Nethermind/Nethermind.Optimism/CL/P2P/OptimismCLP2P.cs

```diff
@@ -84,14 +94,22 @@ public OptimismCLP2P(
             .BuildServiceProvider();
     }
 
-    private async void OnMessage(byte[] msg, CancellationToken token)
+    private void OnMessage(byte[] msg, CancellationToken token)
     {
         try
         {
             if (TryValidateAndDecodePayload(msg, out var payload))
             {
                 if (_logger.IsTrace) _logger.Trace($"Received payload prom p2p: {payload}");
-                await _blocksP2PMessageChannel.Writer.WriteAsync(payload, token);
+                if ((ulong)payload.BlockNumber <= _headNumber)
+                {
+                    return;
```

Comment:
We should either log something or add a comment like in the other similar case:

```suggestion
                    // Old payload. skip
                    return;
```